src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void execute(FruitTransaction transaction) {
+        OperationHandler handler = operationHandlers.get(transaction.getOperation());
+        if (handler != null) {
+            handler.handle(transaction);
+        } else {
+            throw new UnsupportedOperationException("Operation not supported: "
+                    + transaction.getOperation());
+        }
```

Comment:
```suggestion
        if (handler == null) {
         throw new UnsupportedOperationException("Operation not supported: "
                    + transaction.getOperation());  
        } 
        handler.handle(transaction); 
```
Let's avoid if-else construction